src/server.rs

```diff
@@ -300,6 +302,10 @@ impl LsService {
                         Method::Shutdown => {
                             trace!("shutting down...");
                             this.shut_down.store(true, Ordering::SeqCst);
+
+                            // We must respond, otherwise we wouldn't get a request to exit.
```

Comment:
I don't think this comment is necessary and it is also misleading - the specific behaviour is dependent on the client, not on the spec.